### DISCUSSION TITLE
ファクトリパターンによるPlayerInfoクラスの作成

### DISCUSSION THREAD

**Discussion by Hourier** [hengband/hengband#953] [Closed]
耐性情報をどのように扱うかは引き続き議論中の部分もあるかと思います
提案として、添付ファイルのように一貫性のある構造を検討したいです
但しplayer_type構造体を簡単にPlayerInfoクラスへ置き換えられるかは甚だ疑問なので、やるにしても中長期的なリファクタリング作業になると思われます
皆様のご意見を募集します
※ 耐性の話しかしていないので、どこかで不都合があるかもしれない？

![PlayerInfo素案](https://user-images.githubusercontent.com/66951241/116389483-f1f63e00-a857-11eb-8353-231a06239985.png)
[PlayerInfo素案.txt](https://github.com/hengband/hengband/files/6391089/PlayerInfo.txt)

**Comment by shimitei** [comment]
性格はともかく、種族と職業は密接な点が多い（吸血鬼忍者の光源とか）ためそれぞれ独立したクラスとするのは難点がありそうです。
耐性や能力値であればフィルターを通すような処理が適していると思います。

**Comment by Hourier** [comment]
この設計は事実上破棄されました
代替としてPlayerClass及びPlayerRaceクラスを作り、いい感じの設計になってきたので本件はクローズします
(今後はPlayerPersonalityクラスも作った方がいいかもしれない：コンバット＆チャージマン用の特殊処理を押し込むところ)
